src/Runner.Worker/IssueMatcher.cs

```diff
@@ -184,8 +206,9 @@ public IssuePattern(IssuePatternConfig config, TimeSpan timeout)
 
     public sealed class IssueMatch
     {
-        public IssueMatch(IssueMatch runningMatch, IssuePattern pattern, GroupCollection groups, string defaultSeverity = null)
+        public IssueMatch(string sourceText, IssueMatch runningMatch, IssuePattern pattern, GroupCollection groups, string defaultSeverity = null)
         {
+            SourceText = sourceText;
```

Comment:
This allows the original text to "survive" long enough for us to pass it along to the mechanism that logs (`ExecutionContext::Write`)